### DISCUSSION TITLE
Unable to use usePrepareContractWrite() in nextjs v13

### DISCUSSION THREAD

**Discussion by tarushmohindru** [wevm/wagmi#2487] [Q&A]
One of the routes in my app uses the `useAccount()` and it works perfectly, but whenever i try to use `usePrepareContractWrite()` or `useContractWrite()`, i get the following error -
`No QueryClient set, use QueryClientProvider to set one
    at Mint (./src/app/mint/page.js:37:87)`
I have tried wrapping it in `<QueryClientProvider>` from `@tanstack/react-query` but still the same issue.

**Comment by jxom** [comment]
Can you share a minimal reproducible example via a separate repo so we can look further?

Can also check out our official Next.js templates to try and find out what's going wrong: https://github.com/wagmi-dev/create-wagmi/tree/main/templates/next/default

Specifically the [`layout.tsx`](https://github.com/wagmi-dev/create-wagmi/blob/main/templates/next/default/src/app/layout.tsx) and [`providers.tsx`](https://github.com/wagmi-dev/create-wagmi/blob/main/templates/next/default/src/app/providers.tsx) components.

**Reply by tarushmohindru** [reply]
This is how the page.js of that route looks like - 
```import { abi } from "./abi";
import { useState, useEffect} from "react";

export default function Mint() {
  const [expectedAmount, setExpectedAmount] = useState(0);

  const { address, isConnected } = useAccount();

  const { config } = usePrepareContractWrite({
    abi: abi,
    address: "0xe058C0881aC0BF32743b091623708b7FDe75636e",
    functionName: "mint",
    args: [500],
  });

 
  useEffect(() => {
    setExpectedAmount(amount - (amount * 1) / 100);
  }, [amount]);

  const body = () => {
    if (isConnected) {
      return (
       // some frontend ui
      );
    }
  };

  return (
    <WagmiConfig config={wagmiConfig}>
     //some frontend ui
    </WagmiConfig>
  );
}```

This route has no layout.js. I have tried creating that too and wrapping it with all the wrappers. But still, the issue persists

**Reply by tmm** [reply]
wagmi hooks need to be used inside `WagmiConfig`
